File: app/src/main/java/com/fernandez/pablo/la24gnc/View/ItemSpinnerAdapter.java
package com.fernandez.pablo.la24gnc.View;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.fernandez.pablo.la24gnc.Model.EspecificacionProducto;
import com.fernandez.pablo.la24gnc.R;

import java.util.List;

/**
 * Created by pablo on 15/04/2017.
 */

public class ItemSpinnerAdapter extends BaseAdapter{


    private List<EspecificacionProducto> productos;
    private Context context;

    public ItemSpinnerAdapter(Context context, List<EspecificacionProducto> productos) {
        this.productos = productos;
        this.context = context;
    }

    @Override
    public int getCount() {
        return productos.size();
    }

    @Override
    public Object getItem(int position) {
        return productos.get(position);
    }

    @Override
    public long getItemId(int position) {
        return productos.get(position).getCodigo();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View rowView = convertView;

        if (convertView == null) {
            // Create a new view into the list.
            LayoutInflater inflater = (LayoutInflater) context
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            rowView = inflater.inflate(R.layout.list_item_spinner, parent, false);
        }

        // Set data into the view.
        TextView tvNombreProducto = (TextView) rowView.findViewById(R.id.tvItemSpinner);

        EspecificacionProducto producto= this.productos.get(position);

        tvNombreProducto.setText(producto.getDescripcion());

        return rowView;
    }

}

File: app/src/main/java/com/fernandez/pablo/la24gnc/View/Utils/ItemSpinnerAdapter.kt
package com.fernandez.pablo.la24gnc.View.Utils

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.BaseAdapter
import android.widget.TextView

import com.fernandez.pablo.la24gnc.Model.EspecificacionProducto
import com.fernandez.pablo.la24gnc.R

/**
 * Created by pablo on 15/04/2017.
 */

class ItemSpinnerAdapter(private val context: Context, private val productos: ArrayList<EspecificacionProducto>?) : BaseAdapter() {

    override fun getCount(): Int {
        return productos!!.size
    }

    override fun getItem(position: Int): Any {
        return productos!![position]
    }

    override fun getItemId(position: Int): Long {
        return productos!![position].codigo.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var rowView = convertView

        if (convertView == null) {
            // Create a new view into the list.
            val inflater = context
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            rowView = inflater.inflate(R.layout.list_item_spinner, parent, false)
        }

        // Set data into the view.
        val tvNombreProducto = rowView!!.findViewById(R.id.tvItemSpinner) as TextView

        val producto = this.productos!![position]

        tvNombreProducto.text = producto.descripcion

        return rowView
    }

}
